src/create_table_with_clause.c

```diff
@@ -23,3 +25,43 @@ ts_create_table_with_clause_parse(const List *defelems)
 								 create_table_with_clauses_def,
 								 TS_ARRAY_LEN(create_table_with_clauses_def));
 }
+
+Datum
+ts_create_table_parse_chunk_time_interval(WithClauseResult *parsed_options, Oid column_type,
+										  Oid *interval_type)
+{
+	if (parsed_options[CreateTableFlagChunkTimeInterval].is_default == false)
+	{
+		Datum textarg = parsed_options[CreateTableFlagChunkTimeInterval].parsed;
+		switch (column_type)
+		{
+			case INT2OID:
+			{
+				*interval_type = INT2OID;
+				return DirectFunctionCall1(int2in, CStringGetDatum(TextDatumGetCString(textarg)));
+			}
+			case INT4OID:
+			{
+				*interval_type = INT4OID;
+				return DirectFunctionCall1(int4in, CStringGetDatum(TextDatumGetCString(textarg)));
+			}
+			case INT8OID:
+			{
+				*interval_type = INT8OID;
+				return DirectFunctionCall1(int8in, CStringGetDatum(TextDatumGetCString(textarg)));
+			}
+			case TIMESTAMPOID:
+			case TIMESTAMPTZOID:
+			case DATEOID:
+			{
+				*interval_type = INTERVALOID;
+				return DirectFunctionCall3(interval_in,
+										   CStringGetDatum(TextDatumGetCString(textarg)),
+										   InvalidOid,
+										   -1);
+			}
+		}
+	}
+	*interval_type = InvalidOid;
+	return -1;
```

Comment:
Can u refresh my memory, so why -1 and not 0?

Reply:
This is passed in as default value for interval in create_hypertable but we can pass in whatever we want here while the type is InvalidOid.

Reply:
Lets create a static const with a name just so it doesn't look like a magic value.